web/src/filter.ts

```diff
@@ -1502,8 +1526,8 @@ export class Filter {
         return new Filter(terms);
     }
 
-    has_topic(stream_name: string, topic: string): boolean {
-        return this.has_operand("channel", stream_name) && this.has_operand("topic", topic);
+    has_topic(stream_id: string, topic: string): boolean {
+        return this.has_operand("channel", stream_id) && this.has_operand("topic", topic);
     }
```

Comment:
Given that `filter.has_topic` only has one caller outside tests, I think it might be more reasonable to have this accept a `stream_id`  integer, and do the translation to a `string_id_str` internally.

This change would be good for consistency; it's nice if every `stream_id` variable is an integer.

Reply:
Yeah that sounds good, thanks!